ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/query/query.tsx

```diff
@@ -0,0 +1,19 @@
+import fetch from '../fetch'
+
+type Query = {
+  srcs: string[]
+  count: number
+  cql: string
+  facets: string[]
+}
+
+const query = async (q: Query) => {
+  const res = await fetch('./internal/cql', {
```

Comment:
should you add error handling here?

Reply:
I would say any error handling here should be left to the files using this utility method